src/libs/OptionsListUtils.js

```diff
@@ -77,16 +78,17 @@ Onyx.connect({
 });
 
 const lastReportActions = {};
-const allReportActions = {};
+const allSortedReportActions = {};
 Onyx.connect({
     key: ONYXKEYS.COLLECTION.REPORT_ACTIONS,
     callback: (actions, key) => {
         if (!key || !actions) {
             return;
         }
-        allReportActions[key] = actions;
+        const sortedReportActions = ReportActionUtils.getSortedReportActions(_.toArray(actions), true);
         const reportID = CollectionUtils.extractCollectionItemID(key);
-        lastReportActions[reportID] = _.last(_.toArray(actions));
+        allSortedReportActions[reportID] = sortedReportActions;
```

Comment:
We never clean up `allSortedReportActions` so it keeps gathering garbage between sessions. This confused me a lot since I was able to find reportActions that were not in Onyx. This is also a memory leak.

@neil-marcellini 🙏 fix